src/scripts/main.js

```diff
@@ -2,10 +2,28 @@
 
 function waitFor(element, eventName) {
   // write your code here
+  return new Promise((resolve, reject) => {
+    if (!element) {
+      const error = 'Something went wrong';
+
+      reject(error);
+    } else {
+      element.addEventListener(`${eventName}`, (ev) => {
+        resolve(
+          `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
```

Comment:
The message being resolved includes the element's `nodeName` and `id`. Ensure that these properties are available on the element to avoid potential undefined values in the message.